### PR TITLE
feat(components): add el-text component

### PR DESCRIPTION
**Pull request by gimjin**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

Because vitepress import from published element-plus, so el-text doc example not working. The docs only be packaged after published element-plus has el-text.

Locally review, build first, and edit example vue files:
```diff
<template>
  <el-text class="mx-1" size="large">Large</el-text>
  <el-text class="mx-1">Default</el-text>
  <el-text class="mx-1" size="small">Small</el-text>
</template>

+ <script lang="ts" setup>
+ import { ElText } from '../../../dist/element-plus/es/index.mjs'
+ </script>
```

### CODE REVIEW

**Comment by gimjin** [comment]
Create el-text idea by [How to use Element-Plus only, forget inefficiency DIV+CSS?](https://github.com/element-plus/element-plus/discussions/11493)

**Comment by jw-foss** [comment]
@gimjin thank you for the contribution, I will go through your PR and see if this matches our current design principle, we **WILL ACCEPT** this one.

**Comment by jw-foss** [comment]
@gimjin I was wondering if you can reorganize your commits into smaller pieces with `git reset --mixed`, for instance:

- Component implementation
- Testing
- Documentation

Dividing your changes into smaller ones will boost the review process.

**Comment by jw-foss** [comment]
To make this work, you will need to introduce this component in `packages/element-plus/component.ts` file and include it in the default exports.

**Comment by gimjin** [comment]
> @gimjin thank you for the contribution, I will go through your PR and see if this matches our current design principle, we **WILL ACCEPT** this one eventually.

thanks for your guidance @jw-foss , I've adjusted as required.

- [x] reorganize commits into smaller pieces
- [x] el-text in packages/element-plus/component.ts file and include it in the default exports

File: packages/components/text/src/text.ts
Comment (ryuhangyeong) [inline packages/components/text/src/text.ts:9]: ## Question

image

There are more tags. Is there a reason for not supporting it?

Comment (gimjin) [reply packages/components/text/src/text.ts:9]: Yes you are right, HTML Formating should support.

After your reminder, I think of more, such as text related HTML tags:
- default
  - span
- [HTML Formatting](https://www.w3schools.com/html/html_formatting.asp)
  - b
  - strong
  - i
  - em
  - mark
  - ~~small~~ (conflict with type prop)
  - ~~abbr~~ (conflict with el-tooltip)
  - del
  - ins
  - cite
  - kbd
  - samp
  - sub
  - sup
- [HTML Semantics](https://www.w3schools.com/html/html5_semantic_elements.asp)
  - p
  - s
  - h1~6
  - time
  - abbr
- [Functional text](https://www.w3schools.com/tags/default.asp) (find text keywords)
  - pre
  - ~~bdi~~ (better support el-config-provider)
  - ~~bdo~~ (better support el-config-provider)

**I'm concerned, too many enums, confusing element-plus user**. We may want to discuss requirements other than technology, `design principle`, `product direction` or `element-plus user needs`.

Comment (ryuhangyeong) [reply packages/components/text/src/text.ts:9]: In fact, specifying each tag may be unnecessary. So I think it would be better to define it concisely as below and delegate it to the user.

```
  props: {
    as: {
      type: String,
      default: 'span'
    },
  },
```

What do you think?

Comment (gimjin) [reply packages/components/text/src/text.ts:9]: Good idea. like this?

packages/components/text/src/text.ts
```diff
  props: {
    as: {
      type: String,
-     values: ['span', 'p', 'i', 'u', 'del'],
      default: 'span',
    },
  },
```

docs/en-US/component/text.md
```diff
  | Name | Description          | Type                                          | Default |
- | as   | override the element | ^[enum]`'span' \| 'p' \| 'i' \| 'u' \| 'del'` | span    |
+ | as   | override the element | ^[string]                                     | span    |
```

docs/examples/text/override.vue
```vue
<template>
  <el-space direction="vertical">
    <el-text>span</el-text>
    <el-text as="p">This is a paragraph.</el-text>
    <el-text as="b">Bold</el-text>
    <el-text as="i">Italic</el-text>
    <el-text as="sub">Subscripted</el-text>
    <el-text as="sup">Superscripted</el-text>
    <el-text as="ins">Inserted</el-text>
    <el-text as="del">Deleted</el-text>
    <el-text as="mark">Marked</el-text>
  </el-space>
</template>
```
> Exclude em, strong, small from HTML Fromating tags, em as i, strong as b, recommended to use size prop not small tag.

File: packages/theme-chalk/src/text.scss
Comment (ryuhangyeong) [inline packages/theme-chalk/src/text.scss:23]: ## Suggest

line 12 ~ 20 move to here

File: packages/components/text/src/text.ts
Comment (ryuhangyeong) [reply packages/components/text/src/text.ts:9]: good

File: docs/en-US/component/text.md
Comment (ryuhangyeong) [inline docs/en-US/component/text.md:50]:
```
## API
### Attributes
### Slots
```

Reference [here](https://github.com/element-plus/element-plus/blob/dev/docs/en-US/component/link.md)

**Comment by gimjin** [comment]
Already modified code and pushed.

**Comment by gimjin** [comment]
Do I want to synchronize with upstream/dev every push? I have just rebased the latest code.

```bash
git rebase -i upstream/dev
git push -f
```

**Comment by jw-foss** [comment]
> Do I want to synchronize with upstream/dev every push? I have just rebased the latest code.
> 
> ```shell
> git rebase -i upstream/dev
> git push -f
> ```

Technically you don't. Your commits will be squashed and merged onto the HEAD, meaning that you will only have 1 commit in the end, as long as there were no conflicts I don't think you should rebase for every push. BTW, in practical Git, we don't really use `-i` for rebasing the upstream, and for pushing, it is always recommended to use `--force-with-lease` instead of `--force/-f`, considering this is your own feature branch, `-f` is fine, just for making sure you don't accidentally push something to a shared branch, you should do that for every force push.

**Comment by gimjin** [comment]
#11724 PR is before this PR merged dev branch, github robot tip me behind and show MERGE BUTTON, after click the button robot auto create new `merged dev` commit. Because this is not my commit and pollute dev branch history, so run git force rebase.

I already know, only conflict when merging will rebase it.

@jw-foss thanks gift `--force-with-lease`  unfamiliar option.

File: docs/.vitepress/crowdin/en-US/pages/component.json
Comment (ryuhangyeong) [inline docs/.vitepress/crowdin/en-US/pages/component.json:36]: @jw-foss Is the 'promotion' attribute mandatory when adding a new component?

File: packages/components/text/src/text.ts
Comment (ryuhangyeong) [inline packages/components/text/src/text.ts:7]: Add comments.

[reference](packages/components/countdown/src/countdown.ts)

File: packages/theme-chalk/src/text.scss
Comment (ryuhangyeong) [inline packages/theme-chalk/src/text.scss:13]: ## Question

Is there a reason to have a fixed height value? The text area is likely to be long enough.

**Comment by gimjin** [comment]
Done task:
- [x] packages/components/text/src/text.ts add commants
- [x] packages/theme-chalk/src/text.scss remove display and height, now support inline mixed icon and component long text.
- [x] upgrade example icon.vue to mixed.vue

**Review by ryuhangyeong — APPROVED** [review]

File: packages/components/text/src/text.ts
Comment (btea) [inline packages/components/text/src/text.ts:29]: It seems that the default value can be removed.

Comment (gimjin) [reply packages/components/text/src/text.ts]: > It seems that the default value can be removed.

deleted default value.

File: packages/components/text/src/text.ts
Comment (tolking) [inline packages/components/text/src/text.ts:33]: as -> tag ?

refer [tag attributes of Row and Col](https://element-plus.org/en-US/component/layout.html#row-attributes)

Comment (gimjin) [reply packages/components/text/src/text.ts]: as -> tag modified, also modify the docs and unit test.

**Review by btea — APPROVED** [review]

File: docs/en-US/component/text.md
Comment (tolking) [inline docs/en-US/component/text.md:50]:
```suggestion
## API
```

File: docs/en-US/component/text.md
Comment (tolking) [inline docs/en-US/component/text.md:52]:
```suggestion
### Attributes
```

File: docs/en-US/component/text.md
Comment (tolking) [inline docs/en-US/component/text.md:61]:
```suggestion
### Slots
```

File: docs/en-US/component/text.md
Comment (tolking) [inline docs/en-US/component/text.md:65]:
```suggestion
| default | default content |
```

File: packages/components/text/__tests__/text.test.tsx
Comment (tolking) [inline packages/components/text/__tests__/text.test.tsx:20]:
```suggestion
    expect(vm.$el.classList.contains('el-text')).toEqual(true)
    expect(wrapper.text()).toEqual(AXIOM)
```

Comment (tolking) [inline packages/components/text/__tests__/text.test.tsx:20]:
```suggestion
    expect(vm.$el.classList.contains('el-text')).toEqual(true)
    expect(wrapper.text()).toEqual(AXIOM)
```

File: packages/components/text/src/text.vue
Comment (tolking) [inline packages/components/text/src/text.vue:3]: IMO, It feels strange to use template and render function together.

File: docs/en-US/component/text.md
Comment (gimjin) [reply docs/en-US/component/text.md:50]: ElButton supports all experimental features, so I think I'm going to learn button, are you sure I should change?

[reference el-button doc](https://github.com/element-plus/element-plus/blob/dev/docs/en-US/component/button.md)

File: docs/en-US/component/text.md
Comment (gimjin) [reply docs/en-US/component/text.md:52]: Same question as above.

File: docs/en-US/component/text.md
Comment (gimjin) [reply docs/en-US/component/text.md:61]: Same question as above.

File: docs/en-US/component/text.md
Comment (gimjin) [reply docs/en-US/component/text.md:65]: Ok, unified processing after other conversations clarify the answer.

File: packages/components/text/__tests__/text.test.tsx
Comment (gimjin) [reply packages/components/text/__tests__/text.test.tsx:20]: Ok, unified processing after other conversations clarify.

Comment (gimjin) [reply packages/components/text/__tests__/text.test.tsx:20]: Ok, unified processing after other conversations clarify.

File: packages/components/text/src/text.vue
Comment (gimjin) [reply packages/components/text/src/text.vue:3]: Ok, I'll rewrite the template using component tag.

File: docs/en-US/component/text.md
Comment (tolking) [reply docs/en-US/component/text.md:50]: The ElButton page contains two components, but there is only one

Comment (gimjin) [reply docs/en-US/component/text.md:50]: Got it.

**Review by tolking — APPROVED** [review]

**Comment by ryuhangyeong** [comment]
@gimjin nice~

**Comment by gimjin** [comment]
> @gimjin nice~

😸🥂

**Comment by jw-foss** [comment]
@gimjin thank you for your hard working!! I will review the ASAP 👍

File: docs/en-US/component/text.md
Comment (jw-foss) [inline docs/en-US/component/text.md:59]: Maybe we should call it `as`, since most UIs do.

**Review by jw-foss — COMMENTED** [review]
@gimjin good job and excellent work! I have no further comments but one for the API naming, would you mind address it so we can merge this one and get it released!

File: docs/en-US/component/text.md
Comment (tolking) [reply docs/en-US/component/text.md:59]: refer [tag attributes of Row and Col](https://element-plus.org/en-US/component/layout.html#row-attributes)

Comment (gimjin) [reply docs/en-US/component/text.md:59]: At the beginning Reference [chakra-ui](https://vue.chakra-ui.com/text) uses `as` prop name, but el-row/col use `tag` prop name, so change prop name to `tag`. 
`tag` el-row/col used, `as` is most UIs do, difficult choice.

😵 Wait for @jw-foss  and @tolking  discuss the results, I will change it as soon as possible.

Comment (jw-foss) [reply docs/en-US/component/text.md:59]: @gimjin let's keep consistent with `el-col` and `el-row`. Approved! /cc @tolking

**Review by jw-foss — APPROVED** [review]
@gimjin good job and excellent work! I have no further comments but one for the API naming, would you mind address it so we can merge this one and get it released!

**Comment by jw-foss** [comment]
I will merge this later this week, due to the hotfix tonight.

File: packages/components/text/src/text.ts
Comment (ryuhangyeong) [inline packages/components/text/src/text.ts:34]: ## Suggest

Inspired by the link below, it would be nice to improve the tag prop to receive components.

You can do this later!

https://github.com/element-plus/element-plus/pull/11853/files#diff-043025f7e5f7add14be78f47786cc3980c374f9e6d7ba5f02fc9b901b2d5d1cfR110

Comment (tolking) [reply packages/components/text/src/text.ts:34]: The `tag` attribute of the Button component supports incoming components mainly to support `router-link` and `nuxt-link`. This component is not recommended this.

Comment (gimjin) [reply packages/components/text/src/text.ts:34]: I see 👌

**Comment by jw-foss** [comment]
@gimjin thank you for your excellent work and patient ❤️, I just merged this PR! Good job! 🚀

**Comment by gimjin** [comment]
Thank you for review my graffiti code @jw-foss @ryuhangyeong @tolking @btea 🍻.

I have developed ellipsis tooltip and line-clamp in my line-clamp branch, but unit test some exceptions in el-tooltip, I will create PR after fix.

**Pull request merged**
